windows-utils/replace-symlinks.js

```diff
@@ -0,0 +1,94 @@
+const shell = require("shelljs");
+
+async function getSymlinks(dirPath) {
+    let symlinks = [];
+    const path = dirPath
+        .replaceAll("/", "\\")
+        .replaceAll("../", "")
+        .split("\\")
+        .slice(2)
+        .join("\\");
+    const cwd = process.cwd();
+    const changeDir = cwd + path;
+    const fileList = await shellExec("git ls-files -s");
+
+    if (fileList.stdout.includes("120000")) {
+        const output = fileList.stdout.split("\n");
+        symlinks.push(...extractSymlinks(output, changeDir));
+    }
+    shell.cd(cwd);
+    return symlinks;
+}
+const shellExec = (cmd) =>
+    new Promise((resolve) => {
+        shell.exec(cmd, (code, stdout, stderr) => {
+            resolve({ code, stdout, stderr });
+        });
+    });
+
+function extractSymlinks(output, cwd) {
+    const extractedSymlinks = [];
+    for (let i = 0; i < output.length; i++) {
+        const parsedElement = output[i].split(" ");
+        if (parsedElement[0] === "120000") {
+            const relDir = parsedElement[2]
+                .replace("0\t", "")
+                .replaceAll("/", "\\");
+            const fileName = relDir.split("\\").pop();
+            const fileDir = cwd + "\\" + relDir.replace(fileName, "");
```

Comment:
If you do this stuff _before_ converting the slashes in `relDir`, you could just use the `os.path` module for this stuff, and then convert them only once. I suppose `shell.head` (or `shell.cat` maybe) below could also read a POSIX path, so could be `os.path.join(fileDir, fileName)` and generally less throwing around string constants.